core/src/main/scala/chisel3/Module.scala

```diff
@@ -738,12 +739,47 @@ package experimental {
           case (_, Some(parent)) => parent.toRelativeTarget(root).instOf(this.instanceName, name)
           // If root was defined, and there is no parent, the root was not an ancestor.
           case (Some(definedRoot), None) =>
-            throwException(s"Requested .toRelativeTarget relative to ${definedRoot.name}, but it is not an ancestor")
+            throwException(
+              s"Requested .toRelativeTarget relative to ${definedRoot.name}, but it is not an ancestor of $this"
+            )
           // If root was not defined, and there is no parent, return this.
           case (None, None) => getTarget
         }
     }
 
+    /** Returns a FIRRTL ModuleTarget that references this object, relative to an optional root.
+      *
+      * If `root` is defined, the target is a hierarchical path starting from `root`.
+      *
+      * If `root` is not defined, the target is a hierarchical path equivalent to `toAbsoluteTarget`.
+      *
+      * @note If `root` is defined, and has not finished elaboration, this must be called within `atModuleBodyEnd`.
+      * @note The BaseModule must be a descendant of `root`, if it is defined.
+      * @note This doesn't have special handling for Views.
+      */
+    final def toRelativeTargetToHierarchy(root: Option[Hierarchy[BaseModule]]): IsModule = {
+      // If root was defined, and we are it, return this.
+      if (root.map(_.proto).contains(this)) getTarget
```

Comment:
@azidar this is the check that should work that is failing. Is there a correct incantation here...?

Reply:
> theoretically, we should be able to resolve this given the actual target (e.g. the IsModule) datastructure, rather than the hierarchy datastructure.  convert both the root and the this to absoluteTargets, and then inspect those two things to figure out how to represent the relative path. 

Maybe need some sort of helper on IsModule.relativeTo(another: IsModule)

Reply:
todo: see Target.scala in firrtl, use moduleOpt and striphierarchy to cobble something together here